os/kernel/semaphore/sem_destroy.c

```diff
@@ -85,6 +88,26 @@
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+#ifdef CONFIG_APP_BINARY_SEPARATION
+extern sq_queue_t g_sem_list;
+
+static void sem_unregister(FAR sem_t *sem)
+{
+	sem_t *sem_ptr;
+	irqstate_t flags;
+
+	sem_ptr = (sem_t *)sq_peek(&g_sem_list);
```

Comment:
Should we update list in irqsave? I cant think of a scenario now..

Reply:
I think it needs because of global variable, g_sem_list

Reply:
Should we have peek also inside irqsave is was my thought is.